internal/security/authz/authz.go

```diff
@@ -166,21 +166,31 @@ func Get(t testing.TB, dut *ondatra.DUTDevice) (*authzpb.GetResponse, *Authoriza
 		t.Fatalf("Could not connect gnsi %v", err)
 	}
 	resp, err := gnsiC.Authz().Get(context.Background(), &authzpb.GetRequest{})
+	nopolicyexist := 0
 	if err != nil {
-		t.Fatalf("Authz.Get request is failed on device %s: %v", dut.Name(), err)
-	}
-
-	t.Logf("Authz.Get response is %s", prettyPrint(resp))
-	if resp.GetVersion() == "" {
-		t.Errorf("Version is not set in Authz.Get response")
-	}
-	if resp.GetCreatedOn() > uint64(time.Now().UnixMicro()) {
-		t.Errorf("CreatedOn value can not be larger than current time")
+		statusError, _ := status.FromError(err)
+		if statusError.Code() == codes.FailedPrecondition {
+			t.Logf("Expected error FAILED_PRECONDITION seen for authz Get Request.")
+			nopolicyexist = 1
+		} else {
```

Comment:
unsure why 'else' is necessary here.

Reply:
we are making sure that only expected error code which is "FAILED PRECONDITION" (empty policy get operation) is received in which case we return NIL as expected if anything other than FAILED PRECONDITION we will have to error out which is captured in else section.

Reply:
I'd avoid the else here too.

Why not just:
  if <condition you care about> {
    do things
   }
   if <any other error> {
     do other things
   }
   
   Same effect, just  no real need to use else.